docs/src/examples/darcy.md

```diff
@@ -0,0 +1,138 @@
+
+```julia
+smoothness = 2.0
+corr_length = 0.5
+dofs = 50
+
+grf = GRF.GaussianRandomField(
+    GRF.CovarianceFunction(dim, GRF.Matern(smoothness, corr_length)),
+    GRF.KarhunenLoeve(dofs),
+    pts_per_dim,
+    pts_per_dim,
+)
+```
+
+We define a wrapper around the GRF, and as the permeability field must be positive we introduce a domain constraint into the function distribution. 
+
+```julia
+pkg = GRFJL()
+distribution = GaussianRandomFieldInterface(grf, pkg) # our wrapper from EKP
+domain_constraint = bounded_below(0) # make κ positive
+pd = ParameterDistribution(
+    Dict("distribution" => distribution, "name" => "kappa", "constraint" => domain_constraint),
+) # the fully constrained parameter distribution
+```
+
+Henceforth, the GRF is interfaced in the same manner as any other parameter distribution with regards to interface. We sample a reasonably high-probability value from this distribution as a true value (here all degrees of freedom set with ``u_{\mathrm{true}} = -1.5``). We then use the EKP transform function to build the corresponding instance of the ``\kappa_{\mathrm{true}}``.
+
+```julia
+u_true = -1.5 * ones(dofs,1) # the truth parameter
+κ_true = transform_unconstrained_to_constrained(pd, u_true) # builds and constrains the function.
+κ_true = reshape(κ_true, N, N)
+```
+
+We generate the data sample for the truth in a perfect model setting by evaluating the the model here, and observing the flow velocity at a few subsampled points in each dimension (here `obs_ΔN` leads to a ``7 \times 7`` observation grid), and we assume 5% additive observational noise on the measurements.
+
+```julia
+obs_ΔN = 10 
+darcy = Setup_Param(pts_per_dim, obs_ΔN, κ_true) 
+h_2d = solve_Darcy_2D(darcy, κ_true)
+y_noiseless = compute_obs(darcy, h_2d)
+obs_noise_cov = 0.05^2 * I(length(y_noiseless)) * (maximum(κ_true) - minimum(κ_true))
+truth_sample = vec(y_noiseless + rand(rng, MvNormal(zeros(length(y_noiseless)), obs_noise_cov)))
+```
+
+Now we set up the Bayesian inversion algorithm. The prior we have already defined to construct our truth
+
+```julia
+prior = pd
+```
+
+We define some algorithm parameters, here we take ensemble members larger than the dimension of the parameter space
```

Comment:
As a reader, I would like more elaboration as to why we choose ensemble members larger than the dimension of the parameter space.